### PR TITLE
feat: Delete Story Map

### LINKED ISSUE #736: Story Maps: delete story map (draft/published)
As a story map maker, I want to delete my published story map that I am not satisfied with, so I can create a new one easily.

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Delete Story Map button added

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added

### Related Issues
Fixes #736

### CODE REVIEW

File: src/localization/locales/en-US.json [hunk 1]
```diff
@@ -844,6 +844,11 @@
         "home_draft_label": "Draft",
         "home_edit": "Edit",
         "home_default_description": "Use maps, photos, videos, audio recordings and narrative to create an impactful story.",
+        "delete_label": "Delete",
+        "delete_confirm_title": "Delete {{title}} story map?",
+        "delete_confirm_message": "You cannot undo deleting your story map. Deleting your published story map may cause a missing page error to your story map viewers.",
+        "delete_confirm_button":"Delete Story Map",
+        "deleted_story_map": "<strong>{{title}}</strong> is deleted",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:851]:
```suggestion
        "deleted_story_map": "<strong>{{title}}</strong> was deleted",
```

File: src/localization/locales/en-US.json [hunk 2]
```diff
@@ -844,6 +844,11 @@
         "home_draft_label": "Draft",
         "home_edit": "Edit",
         "home_default_description": "Use maps, photos, videos, audio recordings and narrative to create an impactful story.",
+        "delete_label": "Delete",
+        "delete_confirm_title": "Delete {{title}} story map?",
+        "delete_confirm_message": "You cannot undo deleting your story map. Deleting your published story map may cause a missing page error to your story map viewers.",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:849]:
```suggestion
        "delete_confirm_message": "Are you sure you wish to delete the story map? This cannot be undone.,
```
Should match other deletion text.

File: src/localization/locales/en-US.json [hunk 3]
```diff
@@ -844,6 +844,11 @@
         "home_draft_label": "Draft",
         "home_edit": "Edit",
         "home_default_description": "Use maps, photos, videos, audio recordings and narrative to create an impactful story.",
+        "delete_label": "Delete",
+        "delete_confirm_title": "Delete {{title}} story map?",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:848]:
```suggestion
        "delete_confirm_title": "Delete story map {{title}}?",
```

File: src/storyMap/components/UserStoryMap.js [hunk 4]
```diff
@@ -71,6 +74,8 @@ const UserStoryMap = () => {
     )
   );
 
+  const onDeleteSuccess = useCallback(() => navigate('/'), [navigate]);
```
Comment (paulschreiber) [inline src/storyMap/components/UserStoryMap.js:77]: question: does this go back to root? I would expect to go back to the Story Maps page.

**Review by paulschreiber — CHANGES_REQUESTED** [review]

**Review by paulschreiber — APPROVED** [review]

**Comment by DerekCaelin** [comment]
### Tested:

**Creation**
- [x] Created story map
- [x] Filled text box with 5000 characters
- [x] Swapped between image and youtube video
- [x] Create story map with all special characters in the title
- [ ] Create story map with all spaces in the title

**Deletion**
- [x] Delete story map (eng)
- [ ] Delete story map (es) #736 
- [x] deleted story map should be inaccessible to view
- [x] deleted story map should be inaccessible to edit
- [x] can't edit someone else's story map

**Pull request merged**